manifests/charts/istiod-remote/templates/mutatingwebhook.yaml

```diff
@@ -44,6 +44,11 @@ metadata:
     operator.istio.io/component: "Pilot"
     app: sidecar-injector
     release: {{ .Release.Name }}
+{{- if .Values.global.externalIstiod }}
+    externalIstiod: "true"
+  annotations:
+    istiodClusterIDs: {{ .Values.global.istiodClusterIDs | default "'*'" }}
```

Comment:
@stevenctl Just to explain the design here, although I'm open to tweaking this (but preferably in a followup PR)

`istiodClusterIDs` default value:
1. for non-istiodless remotes it defaults to `""`, no istiod should try.
1. for istiodless remotes it defaults to `*`, meaning any istiod should try (permissions and other location preference will still apply in leader election). However users should typically override to specify the expected control plane cluster(s). We do override it in the test framework, and I will update the docs as well.